pkg/web/api/fromproto/pipeline.go

```diff
@@ -20,13 +20,19 @@ import (
 )
 
 func PipelineConfig(in *apiv1.Pipeline_Config) pipeline.Config {
+	if in == nil {
+		return pipeline.Config{}
+	}
 	return pipeline.Config{
 		Name:        in.Name,
 		Description: in.Description,
 	}
 }
 
 func PipelineDLQ(in *apiv1.Pipeline_DLQ) pipeline.DLQ {
+	if in == nil {
+		return pipeline.DefaultDLQ
```

Comment:
This would change the behavior. This line assumes that a `nil` pipeline DLQ means the same as the default DLQ configuration (not really true). That should not be the case, this code should only convert from one type to the other without any logic.

TL;DR: we should return `pipeline.DLQ{}` instead.